tests/unit_tests/main_test.cpp

```diff
@@ -113,7 +113,8 @@ int main(int argc, char** argv) {
                             continue;
 #endif
 #if !defined(ENABLE_CUBLAS_BACKEND) && !defined(ENABLE_CUSOLVER_BACKEND) && \
-    !defined(ENABLE_CURAND_BACKEND) && !defined(ENABLE_ROCBLAS_BACKEND)
+    !defined(ENABLE_CURAND_BACKEND) && !defined(ENABLE_ROCBLAS_BACKEND) &&  \
```

Comment:
why do we check for ROCBLAS and ROCRAND backend here? Does not look correct. Also, why check for cuSOLVER is twice?

Reply:
This explains why I had problems running the tests. Should be fixed